test/common/storagelib.py

```diff
@@ -90,6 +90,9 @@ def add_loopback_disk(self, size=50, name=None):
         self.machine.execute("partprobe '%s'" % dev)
         # right after unmounting the device is often still busy, so retry a few times
         self.addCleanup(self.machine.execute, f"umount {dev} || true; rm $(losetup -n -O BACK-FILE -l {dev}); until losetup -d {dev}; do sleep 1; done", timeout=10)
+        # check if umount and detaching loopback  device were successul.
+        self.addCleanup(self.machine.execute, f"cat /proc/mounts | ! grep '^{dev}'")
+        self.addCleanup(self.machine.execute, f"losetup -a | ! grep '^{dev}'")
```

Comment:
I'm not sure if that's really it -- the `until losetup` line should time out (and thus fail) if the device is busy. If `losetup` exits successfully without detaching the device, we have big trouble, of course.

Reply:
> I'm not sure if that's really it -- the `until losetup` line should time out (and thus fail) if the device is busy. If `losetup` exits successfully without detaching the device, we have big trouble.

Theoretically could cause issue, with busy devices, but I've expected that `losetup -a` should print all devices, so in case the device is  not there, losetup with next empty device should use number `3` not number `4`.  I think losetup is robust enough, but it is question. If can happen, that losetup, will serve as next number `4` and does not show that 3 is occupied.

Reply:
OK, it doesn't hurt much, let's keep it for a while.